tests/metadata_stores/conftest.py

```diff
@@ -428,3 +447,46 @@ def test_something(config_with_truncation):
     # Set and restore config
     with config.use():
         yield config
+
+
+@pytest.fixture(scope="session")
+def s3_endpoint_url() -> Generator[str, None, None]:
+    """Start a moto S3 server on a random free port."""
+    port = find_free_port()
```

Comment:
Specifically with `ThreadedMotoServer` this is not needed because it has native functionality for this: 

```py
server = ThreadedMotoServer(port=0)
server.start()
host, port = server.get_host_and_port()
```

But it doesn't matter.